lib/components/app/app-menu.js

```diff
@@ -53,36 +142,56 @@ class AppMenu extends Component {
     } = this.props
 
     return (
-      <div aria-label='app-menu' className='app-menu' role='navigation'>
-        <DropdownButton
-          aria-label='Application Menu'
-          className='app-menu-button'
-          id='app-menu'
-          noCaret
-          title={(<Icon type='bars' />)}>
-          <MenuItem onClick={this._showRouteViewer}>
-            <Icon type='bus' /> {languageConfig.routeViewer || 'Route Viewer'}
-          </MenuItem>
-          {callTakerEnabled &&
-            <MenuItem onClick={resetAndToggleCallHistory}>
-              <Icon type='history' /> Call History
-            </MenuItem>
-          }
-          {fieldTripEnabled &&
-            <MenuItem onClick={resetAndToggleFieldTrips}>
-              <Icon type='graduation-cap' /> Field Trip
+      <>
+        <div
+          aria-label={this.state.isPaneOpenLeft ? 'Close Menu' : 'Open Menu'}
+          class='app-menu-icon'
+          onClick={this._expandedPane}
+          onKeyDown={this._expandedPane}
+          role='button'
+          tabIndex={0}>
+          <span className={this.state.isPaneOpenLeft ? 'menu-left-x' : 'menu-top-line'} />
+          <span className={this.state.isPaneOpenLeft ? '' : 'menu-middle-line'} />
+          <span className={this.state.isPaneOpenLeft ? 'menu-right-x' : 'menu-bottom-line'} />
+        </div>
+        <SlidingPane
+          from='left'
+          isOpen={this.state.isPaneOpenLeft}
+          onRequestClose={this._expandedPane}
+          width='320px'>
+          <ul className='app-menu'>
+            <MenuItem
+              className='menu-item'
+              onClick={this._showRouteViewer}>
+              <Icon name='bus' />{languageConfig.routeViewer || 'Route Viewer'}
             </MenuItem>
-          }
-          {mailablesEnabled &&
-            <MenuItem onClick={toggleMailables}>
-              <Icon type='envelope' /> Mailables
+            <MenuItem
+              className='menu-item'
+              onClick={this._startOver}>
+              <Icon name='undo' />Start Over
             </MenuItem>
-          }
-          <MenuItem onClick={this._startOver}>
-            <Icon type='undo' /> Start Over
-          </MenuItem>
-        </DropdownButton>
-      </div>
+            {callTakerEnabled &&
+            <MenuItem
+              className='menu-item'
+              onClick={resetAndToggleCallHistory}>
+              <Icon name='history' />Call History
+            </MenuItem>}
+            {fieldTripEnabled &&
+            <MenuItem
+              className='menu-item'
+              onClick={resetAndToggleFieldTrips}>
+              <Icon name='' />Field Trip
+            </MenuItem>}
+            {mailablesEnabled &&
+            <MenuItem
+              className='menu-item'
+              onClick={toggleMailables}>
+              <Icon name='' />Mailables
+            </MenuItem>}
+            {extraMenuItems && this._addExtraMenuItems(extraMenuItems)}
```

Comment:
I think we can remove this check since the method should return nothing if given nothing. Also, it can be refactored to get the menu items from props itself.

```suggestion
            {this._addExtraMenuItems()}
```